### DISCUSSION TITLE
In addition to custom SQL, how do I alias a column?

### DISCUSSION THREAD

**Discussion by lipan1024** [Budibase/budibase#3759] [Help]
I want to keep paging and I want to give the column an alias. What do I do?

**Comment by mjashanks** [comment]
You could create a new formula column, via Budibase. This will not create a column in your database.

Just click on your table (in the left panel, in `Data`) and click "Create column"

**Reply by lipan1024** [reply]
Thanks for your reply, it is a good idea, but it seems that I will lose my changes by clicking "Fetch tables from database".

**Reply by mjashanks** [reply]
That is true. There is no other way, other than custom query :/

You can implement pagination via custom query & parameters:

`select * from Contacts limit {{ limit }} offset {{ offset }}`

Then, in the `design` section, you would need to pass the correct parameters into the query. You might want to calculate these using Javascript.

**Reply by lipan1024** [reply]
That's good. I've thought of it, but I don't know how to do it.

**Reply by lipan1024** [reply]
@mjashanks Sorry,i click the settings button,not a drawer is displayed  
I followed this procedure:https://docs.budibase.com/guides/writing-a-custom-sql-query  
![image](https://user-images.githubusercontent.com/86212306/146159177-09a94007-c553-4b83-a34d-02d2ebb67ee2.png)

**Reply by mjashanks** [reply]
Are you self-hosting? This was a recent issue that was fixed and deployed. Take an update :)

**Reply by lipan1024** [reply]
Thanks. Yes, it's working now.
